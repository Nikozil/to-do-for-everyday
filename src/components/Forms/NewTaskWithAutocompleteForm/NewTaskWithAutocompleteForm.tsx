import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import TaskAutocomplete from '../../../common_components/TaskAutocomplete/TaskAutocomplete';
import { addTask } from '../../../Redux/modules/tasksSlice';
import NewTaskForm from '../NewTaskForm/NewTaskForm';

const NewTaskWithAutocompleteForm: React.FC<PropsType> = ({ duration }) => {
  const dispatch = useDispatch();

  const [focus, setFocus] = useState(false);
  const [filter, setFilter] = useState('');

  const handleSubmit = (task: string) => {
    dispatch(addTask(task, duration));
  };

  return (
    <>
      <NewTaskForm
        handleSubmit={handleSubmit}
        setFocus={setFocus}
        setFilter={setFilter}
      />
      {focus ? <TaskAutocomplete filter={filter} duration={duration} /> : null}
    </>
  );
};

export default NewTaskWithAutocompleteForm;

interface PropsType {
  duration: Duration;
}
