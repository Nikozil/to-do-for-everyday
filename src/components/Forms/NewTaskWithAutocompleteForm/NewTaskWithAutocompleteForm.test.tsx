import React from 'react';

import { render, screen, waitFor } from '../../../utils/tests/test-utils';
import userEvent from '@testing-library/user-event';
import NewTaskWithAutocompleteForm from './NewTaskWithAutocompleteForm';

describe('NewTaskWithAutocompleteForm tests', () => {
  it('rendering', async () => {
    const { asFragment } = render(
      <NewTaskWithAutocompleteForm duration={{ days: 0 }} />
    );

    userEvent.type(
      screen.getByPlaceholderText(/Новая Задача/i),
      'New Task{enter}'
    );

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
