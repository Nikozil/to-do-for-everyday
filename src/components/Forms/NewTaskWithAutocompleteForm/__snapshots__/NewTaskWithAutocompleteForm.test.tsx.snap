// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskWithAutocompleteForm tests rendering 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="mb-1"
    >
      <input
        autocomplete="off"
        class="form-control newTask"
        id="newTask"
        name="newTask"
        placeholder="Новая Задача"
        type="text"
        value="New Task"
      />
    </div>
  </form>
  <div
    class="autocomplete__wrapper"
  >
    <div
      class="autocomplete__container"
    >
      <span
        class="stub"
      >
        Нет выполненых задач
      </span>
    </div>
  </div>
   
</DocumentFragment>
`;
